deepsource/client.go

```diff
@@ -35,14 +36,24 @@ func (c Client) GetToken() string {
 
 // Returns a new GQLClient
 func New() (*Client, error) {
-	// TODO: Figure out better way to resolve host here
-	apiClientURL := fmt.Sprintf("https://api.%s/graphql/", config.Cfg.Host)
+	apiClientURL := fmt.Sprintf("https://api.%s/graphql/", config.DefaultHostName)
+	if isEnterprise() {
+		apiClientURL = fmt.Sprintf("https://%s/api/graphql/", config.EnterpriseAPIEndpoint)
+	}
 	gql := graphql.NewClient(apiClientURL)
 	return &Client{
 		gql: gql,
 	}, nil
 }
 
+// Checks whether the host is an enterprise host
+func isEnterprise() bool {
```

Comment:
We don't need to check this.  The CLI should work based on the hostname provided by the user.

Reply:
This check helps to frame the API Endpoint URL since it has different structure for cloud and enterprise instance.
For cloud, its : `https://api.deepsource.io/` while for enterprise, its `https://enterprise.deepsource.io/api/`

Reply:
Ack, made the corresponding changes.